### ISSUE TITLE
Exile implant z-level lock and the shuttle's check for z-locks are in different systems.

### ISSUE THREAD

**Issue by ghost** [d3athrow/vgstation13#20443]
#### Description of issue
tl;dr, exile implants (and mommi's) use locked_to_z while shuttle datums use cant_leave_zlevel
Should be an easy fix if any of ya'll can tackle flopping locked_to_z into shuttle datums I'd probably break something.

Does not lock - !!Shuttle!!

Locks - Drifting to an edge of map z-level transition
Locks - Drifting to an edge of map z-level transition in a closet
Locks - Drifting to an edge of map z-level transition in a cake
Locks - Drifting to an edge of map z-level transition in a mech
Locks - Teleporter
Locks - Going through the teleporter in a closet
Locks - Going through the teleporter in a mech
Locks - Going through the teleporter in a cake
Locks - Telescience
Locks - Hand tele
Locks - Hand tele in a closet
Locks - Hand tele in a mech

#### Difference between expected and actual behavior
Get a cancelled launch like the nuke disc. I blame the shuttle using cant_leave_zlevel while exile implants and mommi's use locked_to_z.

#### Steps to reproduce
>Exile implant yourself
>Go to roid
>Get on shuttle
>Click send shuttle to station
>Shuttle actually leaves with you on it, the absolute mad vehicle

#### Specific information for locating

I think this is the actual one for the shuttles,
https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/datums/shuttle.dm
	var/list/cant_leave_zlevel = list(
		/obj/item/weapon/disk/nuclear = "The nuclear authentication disk can't be transported on a shuttle.",
		)
//Checks the shuttle for any offending atoms
/datum/shuttle/proc/forbid_movement()
	var/atom/A = linked_area.contains_atom_from_list(cant_leave_zlevel) //code/game/atoms.dm, 243
	if(A)
		return A
	return 0

Meanwhile mommi's and exile implants use locked_to_z which runs map edge transitions, teleports, and contents

code\datums\helper_datums\teleport.dm:231:			if(MM.locked_to_z != FALSE && destination.z != MM.locked_to_z)
code\game\machinery\mommi_spawner.dm:16:	var/locked_to_zlevel = 0 // Whether to lock the spawned MoMMIs to the z-level
code\game\machinery\mommi_spawner.dm:154:		if (locked_to_zlevel)
code\game\machinery\mommi_spawner.dm:156:			M.locked_to_z = T.z
code\game\objects\items\weapons\implants\exile2.dm:34:	I.locked_to_z = ASTEROID_Z
code\game\turfs\turf.dm:219:				if(L.locked_to_z != 0)
code\game\turfs\turf.dm:220:					if(src.z == L.locked_to_z)
code\game\turfs\turf.dm:224:						move_to_z = L.locked_to_z
code\modules\mob\living\living_defines.dm:58:	var/locked_to_z = 0 // Locked to a Z-level if nonzero.
code\modules\mob\living\silicon\mommi\life.dm:27:	if(locked_to_z)
code\modules\mob\living\silicon\mommi\life.dm:272:	if(!locked_to_z)
code\modules\mob\living\silicon\mommi\life.dm:278:	if(current_zlevel.z != locked_to_z)
code\modules\mob\living\silicon\mommi\life.dm:280:		locked_to_z = 0
code\modules\mob\living\silicon\robot\death.dm:79:			mmi.brainmob.locked_to_z = locked_to_z

Does not lock - !!Shuttle!!
![fucking shuttle break](https://user-images.githubusercontent.com/5564796/47571578-5cc00a80-d8e5-11e8-9f28-44cf476b95f6.png)

Locks - Drifting to an edge of map z-level transition
![you find your way back only plays on the initial transition into z-5](https://user-images.githubusercontent.com/5564796/47571613-76615200-d8e5-11e8-85be-d0703d192b89.png)

Locks - Drifting to an edge of map z-level transition in a closet
![locker lock works](https://user-images.githubusercontent.com/5564796/47571657-8c6f1280-d8e5-11e8-9f4f-525012d1d0f5.png)

Locks - Drifting to an edge of map z-level transition in a cake
![cake lock works](https://user-images.githubusercontent.com/5564796/47571664-9133c680-d8e5-11e8-86ba-47dc4ebe3235.png)

Locks - Drifting to an edge of map z-level transition in a mech
![mech lock works](https://user-images.githubusercontent.com/5564796/47571680-98f36b00-d8e5-11e8-8579-69ddb61e0bf8.png)

Locks - Teleporter
![can t port to telcommsat z-3](https://user-images.githubusercontent.com/5564796/47571705-a7da1d80-d8e5-11e8-8028-b14a3ace8b2f.png)

Locks - Going through the teleporter in a closet
![can t container teleport](https://user-images.githubusercontent.com/5564796/47571718-b58fa300-d8e5-11e8-9307-5108e34d4c15.png)

Locks - Going through the teleporter in a mech
![can t mech](https://user-images.githubusercontent.com/5564796/47571725-bc1e1a80-d8e5-11e8-8a73-2374d28c4240.png)

Locks - Going through the teleporter in a cake
![cake exile implant failure](https://user-images.githubusercontent.com/5564796/47571740-c6401900-d8e5-11e8-9243-9b52fb993668.png)

Locks - Telescience
![can t teleport out](https://user-images.githubusercontent.com/5564796/47571767-d22bdb00-d8e5-11e8-9066-0fd2051a07eb.png)

Locks - Hand tele
![hand tele](https://user-images.githubusercontent.com/5564796/47571814-e8399b80-d8e5-11e8-978d-0065f2d79378.png)

Locks - Hand tele in a closet
![can t closet](https://user-images.githubusercontent.com/5564796/47571849-fe475c00-d8e5-11e8-8a69-cb5058448be6.png)

Locks - Hand tele in a mech
![hand tele mech](https://user-images.githubusercontent.com/5564796/47571832-f4255d80-d8e5-11e8-85cb-b724ea4753e4.png)

Remaining consistency issue is "You find your way back" only plays when you move from a non-locked z-level to the z-level you're locked to. While unnecessary, getting feedback when you map edge to a different size would help players understand they're stuck there without the gps they're unlikely to have.
It happens on code\game\turfs\turf.dm:215

A snowflake exile implanter pad that allowed the user to change the z-level it's locked to would be nice since you could unwisely choose the derelict, telcommsat, or another z-level instead.  As an antag tool, exiling people to the derelict would be neato.

[Bug / Fix]
[Easy Fix]
[Feature Loss]
[help wanted]

#### Length of time in which bug has been known to occur

Probably since the mommi lock break ages ago.

#### Server revision

d1dba7a168f34245613f0891083ea5b25e18fd80

**Comment by RealestEstate** [comment]
you forgot bluespace tomatoes
